includes/rest-api.php

```diff
@@ -603,6 +683,80 @@ function pmpro_rest_api_get_checkout_levels( $request ) {
 			return new WP_REST_Response( $r );
 		}
 
+
+		/// ZAPIER TRIGGERS
+		/**
+		 * Function used for testing authentication through Zapier API.
+		 * @since 2.7
+		 */
+		public function validate_me( $request ) {
+
+			$params = $request->get_params();
+
+			if ( is_user_logged_in() ) {
+			  $me = wp_get_current_user()->display_name;
+			} else {
+				$me = false;
+			}
+		
+			wp_send_json_success( array( 'username' => $me ) );
+		}
+
+
+		function pmpro_rest_api_recent_memberships( $request ) {
+			$params = $request->get_params();
+
+			$response_type = isset( $params['response_type'] ) ? sanitize_text_field( $params['response_type'] ) : null;
+			$limit = apply_filters( 'pmpro_trigger_recent_members_limit', 1 );
+
+			// Grab the useful information.
+			global $wpdb;
+			$SQL = "SELECT mu.user_id, u.user_email, u.user_nicename, mu.membership_id, ml.name as membership_name, mu.status, mu.modified FROM $wpdb->pmpro_memberships_users AS mu
+					LEFT JOIN $wpdb->users AS u
+					ON mu.user_id = u.id
+					LEFT JOIN $wpdb->pmpro_membership_levels AS ml
+					ON ml.id = mu.membership_id
+					WHERE mu.status = 'active' 
+					ORDER BY mu.modified DESC 
+					LIMIT " . intval( $limit );
+			$results = $wpdb->get_results( $SQL );
+
+			$id = isset( $results[0]->user_id ) ? intval( $results[0]->user_id ) : 0; 
+
+			// Generate random ID for Zapier and then also add all member information.
+			if ( $response_type == 'json' ) {
+				wp_send_json( array( 'id' => $id, 'results' => $results ) );
+			}
+
+			return new WP_REST_Response( $results );
+
+		}
+
+		/// Get last order added
+		function pmpro_rest_api_recent_orders( $request ) {
+			$params = $request->get_params();
+
+			$response_type = isset( $params['response_type'] ) ? sanitize_text_field( $params['response_type'] ) : null;
+			$limit = apply_filters( 'pmpro_trigger_recent_orders_limit', 1 );
+
+			global $wpdb;
+			$SQL = "SELECT o.id as order_id, o.code, u.ID as user_id, u.user_email, u.user_nicename, o.billing_name, o.billing_street, o.billing_city, o.billing_state, o.billing_zip, o.billing_country, o.billing_phone, o.subtotal, o.tax, o.couponamount, o.total, o.status, o.gateway, o.gateway_environment, o.timestamp FROM $wpdb->pmpro_membership_orders AS o
+					LEFT JOIN $wpdb->users AS u
+					ON o.user_id = u.ID
+					ORDER BY o.timestamp DESC
+					LIMIT " . intval( $limit );
+
+			$results = $wpdb->get_results( $SQL );
```

Comment:
Formatted SQL for easier reading and using wpdb::prepare for including vars in query.

```suggestion
		  $sql = "
				SELECT
					`o`.`id` AS `order_id`,
					`o`.`code`,
					`u`.`ID` AS `user_id`,
					`u`.`user_email`,
					`u`.`user_nicename`,
					`o`.`billing_name`,
					`o`.`billing_street`,
					`o`.`billing_city`,
					`o`.`billing_state`,
					`o`.`billing_zip`,
					`o`.`billing_country`,
					`o`.`billing_phone`,
					`o`.`subtotal`,
					`o`.`tax`,
					`o`.`couponamount`,
					`o`.`total`,
					`o`.`status`,
					`o`.`gateway`,
					`o`.`gateway_environment`,
					`o`.`timestamp`
				FROM `{$wpdb->pmpro_membership_orders}` AS `o`
				LEFT JOIN `{$wpdb->users}` AS `u`
					ON `o`.`user_id` = `u`.`ID`
				ORDER BY
					`o`.`timestamp` DESC
				LIMIT %d
			";
			
			$results = $wpdb->get_results( $wpdb->prepare( $sql, $limit ) );
```